workshops/pong_game/README.md

```diff
@@ -0,0 +1,213 @@
+  * Get the [`vector`](http://www.grantjenks.com/docs/freegames/api.html#freegames.vector) functions from the `freegames` library. The [freegames library](http://www.grantjenks.com/docs/freegames/index.html) is a basic module with built-in functions that do standard game tasks like moving around and setting a playing field.
+
+# Building The Game
+
+### 1. Generate Random Values
+```python
+def value():
+    return (3 + random() * 2) * choice([1, -1])
+```
+`value` function will return us randomly generate value between `(-5, -3)` or `(3, 5)` each time. This will give **unpredictable direction** to ball.
+
+### 2. Declare Global Variables
+
+Let's start declaring our global variable, we will start with `ball`:
+```python
+ball = vector(0, 0)
+```
+`ball` sets the **position** of tennis like a pong ball at the center of the screen.
+
+Now, we will create `aim` variable which will get randomly generated values by `value` function and use it as parameters of direction and speed for `vector` method: 
+```python
+aim = vector(value(), value())
+```
+The `state` variable will act as a **JSON object** with the two keys being `1` and `2`. This will be used to update the movement of the paddle in the `y-axis`.
+```python
+state = {1: 0, 2: 0}
+```
+
+### 3. Moving the Paddles
+ 
+We want both Paddle for Player **A** and **B** to be able to move Upward or Downward.  We're going to create a new function called `move` that will allow the Paddles to move.
+```python
+def move(player, change):
+    "Move player position by change."
+    state[player] += change
+```
+### 4. Drawing our Paddle
+
+The turtle will be used to create a paddle. It is quite easy and fast to make closed graphical objects in python by using turtle. To create a paddle or simply rectangle follow the code:
+
+```python
+def rectangle(x, y, width, height):
+    "Draw a rectangle at (x, y) with given width and height."
+    up()
+    goto(x, y)
+    down()
+    begin_fill()
+    for count in range(2):
+        forward(width)
+        left(90)
+        forward(height)
+        left(90)
+    end_fill()
+```
+
+What to understand from this code:
+* We declare **rectangle** function with **x**, **y**, **width**, **height** as parameters for the function.
+* **up()** method is used to pull the turtle's pen up from the screen. It does not leave any marks or traces of its movements.
+* The turtle will go to `x`, `y` coordinates using the `goto()` method.
+* **down()** method will put the turtle's pen down.
+* **begin_fill()** tells the turtle that all upcoming closed graphical objects need to be filled with black color.
+* We define a **for loop** to create a rectangle of given `height` and `width`.
+* At last **end_fill()** tells turtle to stop the filling.
+
+![understanding gif](https://cloud-bbd2e31l2.vercel.app/0understanding_gif.gif)
+
+### 5.Draw game and move Pong Ball
+
+```python
```

Comment:
this is a huge chunk of code, and I will suggest you to break it down and explain it step by step

Reply:
The complete block of code is a complete function. If I break it down into pieces, some people will not able to understand the **indentation**. If we are miss placing a single space the code will not work.